src/main/java/mate/academy/springboot/datajpa/controller/CategoryController.java

```diff
@@ -0,0 +1,56 @@
+package mate.academy.springboot.datajpa.controller;
+
+import javax.validation.Valid;
+import mate.academy.springboot.datajpa.model.Category;
+import mate.academy.springboot.datajpa.model.dto.request.CategoryRequestDto;
+import mate.academy.springboot.datajpa.model.dto.response.CategoryResponseDto;
+import mate.academy.springboot.datajpa.service.CategoryService;
+import mate.academy.springboot.datajpa.service.mapper.RequestDtoMapper;
+import mate.academy.springboot.datajpa.service.mapper.ResponseDtoMapper;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/categories")
+public class CategoryController {
+    private final CategoryService categoryService;
+    private final RequestDtoMapper<CategoryRequestDto, Category> requestMapper;
+    private final ResponseDtoMapper<CategoryResponseDto, Category> responseMapper;
+
+    public CategoryController(CategoryService categoryService,
```

Comment:
You can use Lombok annotation to generate constructors:
https://mate-academy.github.io/jv-program-common-mistakes/java-spring-boot/spring-data-jpa/jv-springboot-data-jpa.html#lombok-in-case-you-use-it